kotlin/samples/hello-workflow/src/main/java/com/squareup/sample/helloworkflow/HelloWorkflowActivity.kt

```diff
@@ -30,6 +30,7 @@ class HelloWorkflowActivity : AppCompatActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     runner = setContentWorkflow(viewRegistry, HelloWorkflow, savedInstanceState)
```

Comment:
I'm not thrilled with this extension method any more, but have no change to suggest for this PR. I'll follow up after you merge.

Reply:
Pass an optional (default true) parameter `startAutomatically` so you only have to tack on this extra `startWorkflow` call if you care about outputs? That feels kinda hacky…

Reply:
I'm also not a fan of the way this works right now because it means if there's any period of time when there are no activities/fragments subscribed, outputs sent during that time will be dropped.

Reply:
I _think_ what I kind of want is a multicasting operator that buffers items _when there are zero subscribers_, and is just a pass-through `publish`-type broadcast when there are _one or more subscribers_. Basically just ensures that _someone_, at least one observer, sees every element. But that's tricky to reason about, because if I'm subscribing to this stream, whether or not I miss an item can be decided anywhere else in the code by whether or not someone else beats me to subscribing or not.

Leaving outputs as a channel doesn't solve the multicasting problem, but just abdicates responsibility for even dealing with it. So it just pushes it downstream to `WorkflowRunner` which, if we want to keep it Rx-only, still needs to multicast that channel to turn it into an `Observable`.

Reply:
Maybe `WorkflowRunner` is the better place to solve this though. `WorkflowRunner` completely owns the `WorkflowHost`, so doesn't need to worry about arbitrary multicasting. If `WorkflowHost` exposes outputs as a `ReceiveChannel`, `WorkflowRunner` can require a statically-configured handler for outputs, instead of exposing them as a stream. So when you call `setContentWorkflow` (or `WorkflowRunner.of`), you have to pass in a function that handles outputs. This is probably a terrible idea because it would be super easy to accidentally close over your `Activity`/`Fragment` instance and create memory leaks, or at least end up trying to interact with dead components if outputs are emitted at the wrong time.

Reply:
`WorkflowLayout` could help with this. Since it knows when it's attached or not, it could be responsible for backpressure when nothing's attached. So `WorkflowRunner` could expose a `ReceiveChannel` of outputs, and `setContentWorkflow` could take the output handler function and pass it to `WorkflowLayout`. `WorkflowLayout`'s internal `Coordinator` could then subscribe to the channel only when there's a _valid_ observer.

Reply:
Making `outputs` a non-broadcast channel again would mean that we could drop the `start` method and tie `WorkflowHost` cancellation to the outputs channel, like it was to the `updates` channel before.

Reply:
Chatted about this offline, summarizing in #430.